ruby/rspec/tutorial.md

```diff
@@ -0,0 +1,154 @@
+# RSpec Tutorial/Guide
+
+RSpec is a testing tool for Ruby, it is the most frequently used testing library in Ruby applications, and it was the source of inspiration for testing tools in other languages like jasmine and mocha.  This tutorial will aim to steer you into the path of testing and show you the different ways of testing.
+
+There are some key parts in an RSpec test:
+* `describe`: it is an example group, usually groups the tests of a method or class.
+* `it`: the atom of the tests, it should contain one test/expectation.
+* `before`: hook that executes before a  group of tests, depending where is located, if it’s located inside a describe or context, it will be executed before  the tests inside the describe or context. By default runs before each test, but you can pass the option `:all` to run before all of them only.  It’s usually used for testing setup
+* `after`: same as before, but after the tests, it’s usually used for teardown setup.
+* `context`: same behavior as `describe` but it’s used for a semantic purpose: each block of tests inside the context behaves in a certain way or under certain conditions.
```

Comment:
a short example would be helpful, `context` is often confusing for begginers

Reply:
In the model spec below, there is an example, maybe I can tell to check that one, what do you think?

Reply:
You are right, I think this example is clear enough